experiment/run_experiment.py

```diff
@@ -242,15 +266,114 @@ def copy_resources_to_bucket(config_dir: str, config: Dict):
 
     def __init__(self, config: Dict):
         self.config = config
         self.instance_name = experiment_utils.get_dispatcher_instance_name(
             config['experiment'])
         self.process = None
 
+    def create_async(self):
+        """Noop in local experiments."""
+
+    def start(self):
+        """Start the experiment on the dispatcher."""
+        shared_volume_dir = os.path.abspath('shared-volume')
+        if not os.path.exists(shared_volume_dir):
+            os.mkdir(shared_volume_dir)
+        shared_volume_volume_arg = '{0}:{0}'.format(shared_volume_dir)
+        shared_volume_env_arg = 'SHARED_VOLUME={}'.format(shared_volume_dir)
+        sql_database_arg = 'SQL_DATABASE_URL=sqlite:///{}'.format(
+            os.path.join(shared_volume_dir, 'local.db'))
+
+        home = os.environ['HOME']
+        host_gcloud_config_arg = (
+            'HOST_GCLOUD_CONFIG={home}/{gcloud_config_dir}'.format(
+                home=home, gcloud_config_dir='.config/gcloud'))
+
+        base_docker_tag = experiment_utils.get_base_docker_tag(
+            self.config['cloud_project'])
+        set_instance_name_arg = 'INSTANCE_NAME={instance_name}'.format(
+            instance_name=self.instance_name)
+        set_experiment_arg = 'EXPERIMENT={experiment}'.format(
+            experiment=self.config['experiment'])
+        set_cloud_project_arg = 'CLOUD_PROJECT={cloud_project}'.format(
+            cloud_project=self.config['cloud_project'])
+        set_cloud_experiment_bucket_arg = (
+            'CLOUD_EXPERIMENT_BUCKET={cloud_experiment_bucket}'.format(
+                cloud_experiment_bucket=self.config['cloud_experiment_bucket']))
+        docker_image_url = '{base_docker_tag}/dispatcher-image'.format(
+            base_docker_tag=base_docker_tag)
+        volume_arg = '{home}/.config/gcloud:/root/.config/gcloud'.format(
+            home=home)
+        command = [
+            'docker',
+            'run',
+            '-ti',
+            '--rm',
+            '-v',
+            volume_arg,
+            '-v',
+            '/var/run/docker.sock:/var/run/docker.sock',
+            '-v',
+            shared_volume_volume_arg,
+            '-e',
+            shared_volume_env_arg,
+            '-e',
+            host_gcloud_config_arg,
+            '-e',
+            set_instance_name_arg,
+            '-e',
+            set_experiment_arg,
+            '-e',
+            set_cloud_project_arg,
+            '-e',
+            sql_database_arg,
+            '-e',
+            set_cloud_experiment_bucket_arg,
+            '-e',
+            'LOCAL_EXPERIMENT=True',
+            '--cap-add=SYS_PTRACE',
+            '--cap-add=SYS_NICE',
+            '--name=dispatcher-container',
+            docker_image_url,
+            '/bin/bash',
+            '-c',
+            'echo ${CLOUD_EXPERIMENT_BUCKET}/${EXPERIMENT}/input && '
+            'gsutil -m rsync -r '
+            '"${CLOUD_EXPERIMENT_BUCKET}/${EXPERIMENT}/input" ${WORK} && '
```

Comment:
This means this requires an experiment to have already happened and these stuff built and then local experiment uses it. i dont think we can call this local experiment then. local experiment should be end-to-end building of things.

Reply:
>This means this requires an experiment to have already happened and these stuff built and then local experiment uses it.

It doesn't mean that. The directory contains things that were synced earlier by this file (which runs on the host).